### PR TITLE
Add js/rust bindings for inline source map generation

### PR DESCRIPTION
**Pull request by vstefanovic97**
Fixes https://github.com/embroider-build/content-tag/issues/61

Right now I just added an extra argument to make it a non-braking change, but if you feel like we want to have second argument be an object that accepts the `filename: string` and `input_source_maps: bool`I can also change it to that

### CODE REVIEW

**Comment by ef4** [comment]
Thanks.

I think the suggestion of changing to an options object and accepting a breaking change is good, please update to that version.

**Comment by vstefanovic97** [comment]
@ef4 this is my first time reading and writting rust so I gave it my best shot, as far as I could tell it doesn't support to have an optional object as an argument in rust, so I took care of that in js space and make sure to always provide all arguments with defaults to the rust based function.

As I don't have knowledge of rust code might need cleanup from someone who actually know what they are doing, but at least I ran all tests and it seems to be working :)

Let me know if you have any feedback

File: pkg/standalone.js [hunk 1]
```diff
@@ -1,5 +1,25 @@
-import init from "./standalone/content_tag.js";
+import init from "./node/content_tag.js";
```
Comment (NullVoxPopuli) [inline pkg/standalone.js:1]: we don't want to use the `node` version of this because it's not at all compatible with browsers.

Likewise (and most annoyingly!) the standalone version also isn't compatible with node (I have a small hunch though that the standalone version _will eventually_ be compatible with node... but we'll see)

File: Cargo.toml [hunk 2]
```diff
@@ -35,6 +35,7 @@ lazy_static = "1.4.0"
 base64 = "0.21.4"
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
+serde-wasm-bindgen = "0.4"
```
Comment (NullVoxPopuli) [inline Cargo.toml:38]: why this change?

File: pkg/standalone.js (hunk 1 above)
Comment (NullVoxPopuli) [reply pkg/standalone.js:1]: the way you can test this is by booting up Vite (`pnpm start`) and see if the module loads and parses, etc.

(This is probably also a good way to spot check sourcemaps)

**Review by NullVoxPopuli — CHANGES_REQUESTED** [review]
Left some comments about the usage of `node`'s wasm output in the `standalone` package :upside_down_face:

File: pkg/standalone.js (hunk 1 above)
Comment (vstefanovic97) [reply pkg/standalone.js]: Ahh my mistake, not sure why I did that probably copy/paste mistake when I was duplicating files.
It should be fixed now @NullVoxPopuli

File: Cargo.toml (hunk 2 above)
Comment (vstefanovic97) [reply Cargo.toml:38]: I use it's `from_value` function to parse the object argument [here](https://github.com/embroider-build/content-tag/pull/62/files#diff-c030113407a164cbd1ad3500c613f59abd4346ba13ee5b653c56f393c937b0a8R15)

Comment (NullVoxPopuli) [reply Cargo.toml:38]: is it a successor to regular wasm-bindgen? or are the non-serde deps just on their way out anyway?

Comment (vstefanovic97) [reply Cargo.toml:38]: image
I got this from the docs, that's why I used it https://docs.rs/serde-wasm-bindgen/latest/serde_wasm_bindgen/.

Comment (vstefanovic97) [reply Cargo.toml:38]: tbh I didn't really have time to learn wasm or rust enough to properly understand what I was doing

File: pkg/standalone.js (hunk 1 above)
Comment (NullVoxPopuli) [reply pkg/standalone.js]: thanks!!

File: Cargo.toml (hunk 2 above)
Comment (NullVoxPopuli) [reply Cargo.toml:38]: oooo that's nice, thanks for providing that info!

**Review by NullVoxPopuli — APPROVED** [review]
lgtm

File: test/parse.test.js [hunk 3]
```diff
@@ -173,7 +173,7 @@ describe(`parse`, function () {
       p.process(
         `const thing = "face";
   <template>Hi`,
-        "path/to/my/component.gjs",
```
Comment (NullVoxPopuli) [inline test/parse.test.js:176]: is the old way still able to be supported?

Comment (ef4) [reply test/parse.test.js:176]: Upthread I said a breaking change is OK, so I don't think we should keep the old way. It's a very easy upgrade.

Comment (NullVoxPopuli) [reply test/parse.test.js:176]: excellent!

**Review by mansona — APPROVED** [review]
Looks good to me 👍

**Comment by vstefanovic97** [comment]
@ef4 does it look good to you, can this get merged/released?
